domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/main/kotlin/ch/tutteli/atrium/domain/robstoll/lib/creating/pathAssertions.kt

```diff
@@ -1,18 +1,343 @@
+    val failureDueToAccessDeniedHint = ExpectImpl.builder.explanatory
+        .withExplanation(FAILURE_DUE_TO_ACCESS_DENIED)
+        .build()
+    return try {
+        val hints = hintForOwnersAndPermissions(path)
+        hints.add(0, failureDueToAccessDeniedHint)
+        ExpectImpl.builder.explanatoryGroup.withDefaultType
+            .withAssertions(hints)
+            .build()
+    } catch (e: IOException) {
+        failureDueToAccessDeniedHint
+    }
+}
+
+
+private fun hintForFileNotFound(path: Path) =
+    ExpectImpl.builder.explanatoryGroup.withDefaultType
+        .withAssertions(
+            hintForNoSuchFile(),
+            hintForClosestExistingParent(path)
+        )
+        .build()
+
+private fun findHintForProblemWithParent(path: Path): Assertion? {
+    val absolutePath = path.toAbsolutePath()
+    var currentParentPart = absolutePath.root
+    for (part in absolutePath) {
+        currentParentPart = currentParentPart.resolve(part)
+        if (currentParentPart != path) {
+            try {
+                val attributes = currentParentPart.readAttributes<BasicFileAttributes>()
+                if (!attributes.isDirectory) {
+                    return hintForParentFailure(
+                        currentParentPart,
+                        explanation = hintForNotDirectory(attributes.fileType)
+                    )
+                }
+            } catch (e: AccessDeniedException) {
+                return hintForParentFailure(
+                    currentParentPart.parent,
+                    explanation = hintForAccessDenied(currentParentPart.parent)
+                )
+            } catch (e: IOException) {
+                return hintForParentFailure(
+                    currentParentPart,
+                    explanation = hintForFileSpecificIoException(currentParentPart, e)
+                )
+            }
+        }
+    }
+    return null
+}
+
+private fun hintForNoSuchFile() =
+    ExpectImpl.builder.explanatory
+        .withExplanation(FAILURE_DUE_TO_NO_SUCH_FILE)
+        .build()
+
+private fun hintForNotDirectory(actualType: Translatable) =
+    ExpectImpl.builder.explanatory
+        .withExplanation(FAILURE_DUE_TO_WRONG_FILE_TYPE, actualType, A_DIRECTORY)
+        .build()
+
+private fun hintForParentFailure(parent: Path, explanation: Assertion) =
+    ExpectImpl.builder.explanatoryGroup.withDefaultType
+        .withAssertions(
+            ExpectImpl.builder.descriptive.failing
+                .withDescriptionAndRepresentation(FAILURE_DUE_TO_PARENT, parent)
+                .build(),
+            when (explanation) {
+                is AssertionGroup -> explanation
+                else -> ExpectImpl.builder.explanatoryGroup.withDefaultType
+                    .withAssertion(explanation)
+                    .build()
+            }
+        ).build()
+
+private fun hintForOtherIoException(exception: IOException) =
+    ExpectImpl.builder.explanatory
```

Comment:
Thanks for the tip! I have created a constant `IO_EXCEPTION_STACK_TRACE_LENGTH` for the maximum reported stack trace. I dislike this approach because it is essentially hardcoded, but I also do not see a nice way to expose this to the API to be changeable. Thoughts?

Reply:
I suggest we address this in a different issue, I have created #234